src/lib/server/cms.ts

```diff
@@ -59,215 +108,177 @@ export async function cmsFromContent(
-						autoplay: Number(match.groups?.autoplay),
-						raw: match[0]
-					});
-					break;
-				case 'tagWidget':
-					tagSlugs.add(match.groups.slug);
-					tokens.push({
-						type: 'tagWidget',
-						slug: match.groups.slug,
-						display: match.groups?.display,
-						raw: match[0]
-					});
-					break;
-				case 'specificationWidget':
-					specificationSlugs.add(match.groups.slug);
-					tokens.push({
-						type: 'specificationWidget',
-						slug: match.groups.slug,
-						raw: match[0]
-					});
-					break;
-				case 'pictureWidget':
-					pictureSlugs.add(match.groups.slug);
-					// With multiple options, to handle any ordering for the options, we need to parse the string again
-					const raw = match[0];
-					const fit = /[?\s]fit=(?<fit>(cover|contain))/.exec(raw)?.groups?.fit as
-						| 'cover'
-						| 'contain'
-						| undefined;
-					const width = /[?\s]width=(?<width>\d+)/.exec(raw)?.groups?.width;
-					const height = /[?\s]height=(?<height>\d+)/.exec(raw)?.groups?.height;
-					tokens.push({
-						type: 'pictureWidget',
-						slug: match.groups.slug,
-						raw,
-						fit,
-						width: width ? Number(width) : undefined,
-						height: height ? Number(height) : undefined
-					});
-					break;
-				case 'contactFormWidget':
-					contactFormSlugs.add(match.groups.slug);
-					tokens.push({
-						type: 'contactFormWidget',
-						slug: match.groups.slug,
-						raw: match[0]
-					});
-					break;
-				case 'countdownWidget':
-					countdownFormSlugs.add(match.groups.slug);
-					tokens.push({
-						type: 'countdownWidget',
-						slug: match.groups.slug,
-						raw: match[0]
-					});
-					break;
-				case 'tagProducts':
-					tagProductsSlugs.add(match.groups.slug);
-					tokens.push({
-						type: 'tagProducts',
-						slug: match.groups.slug,
-						display: match.groups?.display,
-						raw: match[0]
-					});
-					break;
-				case 'galleryWidget':
-					gallerySlugs.add(match.groups.slug);
-					tokens.push({
-						type: 'galleryWidget',
-						slug: match.groups.slug,
-						display: match.groups?.display,
-						raw: match[0]
-					});
-					break;
+	const orderedMatchesMobile = [
+		...matchAndSort(mobileContent, PRODUCT_WIDGET_REGEX, 'productWidget'),
+		...matchAndSort(mobileContent, CHALLENGE_WIDGET_REGEX, 'challengeWidget'),
+		...matchAndSort(mobileContent, SLIDER_WIDGET_REGEX, 'sliderWidget'),
+		...matchAndSort(mobileContent, TAG_WIDGET_REGEX, 'tagWidget'),
+		...matchAndSort(mobileContent, SPECIFICATION_WIDGET_REGEX, 'specificationWidget'),
+		...matchAndSort(mobileContent, CONTACTFORM_WIDGET_REGEX, 'contactFormWidget'),
+		...matchAndSort(mobileContent, PICTURE_WIDGET_REGEX, 'pictureWidget'),
+		...matchAndSort(mobileContent, COUNTDOWN_WIDGET_REGEX, 'countdownWidget'),
+		...matchAndSort(mobileContent, TAG_PRODUCTS_REGEX, 'tagProducts'),
+		...matchAndSort(mobileContent, GALLERY_WIDGET_REGEX, 'galleryWidget')
+	].sort((a, b) => (a.index ?? 0) - (b.index ?? 0));
```

Comment:
This is not what I meant

You still do this if `mobileContent` is an empty string or undefined.

It's really easy, you can make `processMatches` take a string as argument, and create the ordered matches inside `processMatch`.

And only call `processMatches` for mobile if `mobileContent` is defined